src/main/java/org/spongepowered/api/CatalogType.java

```diff
@@ -25,24 +25,41 @@
 package org.spongepowered.api;
 
 /**
- * Represents a type of a catalog that can be used to identify types
- * without using an {@link Enum}.
+ * Represents a type of a catalog that can be used to identify types without
+ * using an {@link Enum}.
+ *
+ * <p>Implementing classes must meet the following requirements for all its
+ * members, including those which are not listed in the classes listed in the
+ * {@link org.spongepowered.api.util.annotation.CatalogedBy CatalogedBy}
+ * annotation:
+ * <ul>
+ * <li>Every pair (a, b) that matches:
+ * <ul>
+ * <li>a.equals(b) must also match a == b</li>
+ * <li>a.getId().equalsIgnoreCase(b.getId()) must also match a == b</li>
+ * </ul>
+ * </li>
+ * </ul>
+ * </p>
  */
 public interface CatalogType {
 
     /**
-     * Gets the unique identifier of this {@link CatalogType}. The identifier
-     * can be formatted however needed.
+     * Gets the unique identifier of this {@link CatalogType}. The identifier is
+     * case insensitive, thus there cannot be another instance with a different
+     * character case. The id of this instance must remain the same as long as
+     * the server is running. The identifier can be formatted however needed.
```

Comment:
Change this to 'must remain the same for the entire duration of its existence'. 'As long as the server is running' means something else entirely.